### ISSUE TITLE
False-positive N803 with overrides

### ISSUE THREAD

**Issue by bersbersbers** [astral-sh/ruff#15925]
### Description

This is my code. `BadClass` is bad, but I have to live with it:

```python
from typing import override


class BadClass:
    def fun(self, DaTa: int) -> None: ...  # noqa: N803


class MyClass(BadClass):
    @override
    def fun(self, DaTa: int) -> None: ...
```

`ruff --isolated check --select=N803 bug.py` (v0.9.4) gives:

```
>ruff --isolated check --select=N803 bug.py
bug.py:10:19: N803 Argument name `DaTa` should be lowercase
   |
 8 | class MyClass(BadClass):
 9 |     @override
10 |     def fun(self, DaTa: int) -> None: ...
   |                   ^^^^^^^^^ N803
   |
```

However, I'd argue I have to name that argument that way - otherwise, `pyright` tells me:
```
c:\Git\project\bug.py
  c:\Git\project\bug.py:10:9 - error: Method "fun" overrides class "BadClass" in an incompatible manner
    Parameter 2 name mismatch: base parameter is named "DaTa", override parameter is named "data" (reportIncompatibleMethodOverride)
1 error, 0 warnings, 0 informations
```

So I'd argue that N803 should ignore functions with `@override` (as well as their `@overload`s!).

**Comment by MichaReiser** [comment]
Thanks, skipping `N803` for overriden methods does make sense to me.

**Comment by VascoSch92** [comment]
Regarding as well as their @overloads!, I believe this is not possible since the linter only has access to one script at a time. This means that if the base method is in a different script from its overloads, it cannot verify that one is overriding the other. (@MichaReiser, is that correct? 🙂)

That said, skipping N803 for overridden methods seems like a good idea.

**Comment by MichaReiser** [comment]
Yes, Ruff can't check today if the decorated method actually overrides a method (in the correct way) from a parent class but we can just trust the developer here. 

We plan to implement a rule similar to pyright's for Red Knot, our upcoming type checker.
